wp/plugin/lib/wp_utils.ml

```diff
@@ -53,22 +52,38 @@ let clear_mapper : Term.mapper = object
     super#map_term cls t'
 end
 
-(* Reads in the program_t and its architecture from a file. *)
-let read_program (ctxt : ctxt) ~(loader : string) ~(filepath : string)
-  : Program.t * Theory.target =
-  let mk_digest = Cache.Digests.get_generator ctxt ~filepath ~loader in
-  let program_digest = Cache.Digests.program mk_digest in
-  match Cache.Program.load program_digest with
-  | Some prog ->
+(* FIXME: read context to determine whether we need the "fat" project or not. *)
```

Comment:
What is the plan for determining whether a project should be skinny or not?

Reply:
It's whether or not the `init-mem` flag is set.